shared/src/native-src/string.h

```diff
@@ -13,48 +14,68 @@
 #define WStrLen(value) (size_t) std::char_traits<WCHAR>::length(value)
 #endif
 
-namespace shared {
+namespace shared
+{
+std::string PadLeft(const std::string& txt, std::size_t len, char c = ' ');
 
-    typedef std::basic_string<WCHAR, std::char_traits<WCHAR>, std::allocator<WCHAR>> WSTRING;
+inline std::string Hex(unsigned long value, int padding = 8, std::string prefix = "0x")
+{
+    std::stringstream str;
+#ifdef _WIN32
+    str << prefix << std::hex << std::uppercase << std::right << std::setfill('0') << std::setw(padding) << value;
+#else
+    std::stringstream h;
+    h << std::hex << std::uppercase << value;
+    str << prefix << PadLeft(h.str(), padding, '0');
+#endif
+    return str.str();
```

Comment:
why not using the same code between linux and windows ?
I've tested the l25 on linux and it's the same result on linux.

Reply:
It is the same issue as the PadLeft. When I programmed it, `setfill` didn't work properly in Linux. But I'll give it another try now.